### PR TITLE
Add kubeadm 1.9 upgrade docs

### PR DESCRIPTION
**Pull request by jamiehannaford**
Provides explicit instructions on how to upgrade to 1.9. Much of this was copy pasta'd from our 1.8 docs, so if there's a better way to document this I'm open to alternatives.

/cc @luxas

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6485)

### CODE REVIEW

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` Kubernetes cluster running version 1.8.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md) carefully.
+- `kubeadm upgrade` now allows you to upgrade etcd. To do so, use the appropriate flag.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up what's important to you. For example, any app-level state, such as a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
+
+Also, note that only one minor version upgrade is supported. That is, you can only upgrade from, say 1.8 to 1.9, not from 1.7 to 1.9.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Also, note that only one minor version upgrade is supported. For example, you can only upgrade from 1.8 to 1.9, not from 1.7 to 1.9.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Upgrading your control plane
+
+You have to carry out the following steps by executing these commands on your master node:
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Execute these commands on your master node:

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+```shell
+$ export VERSION=$(curl -sSL https://dl.k8s.io/release/stable.txt) # or manually specify a released Kubernetes version
+$ export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm
+$ chmod a+rx /usr/bin/kubeadm
+```
+
+**Caution:** Upgrading the `kubeadm` package on your system prior to upgrading the control plane causes a failed upgrade. 
+Even though `kubeadm` is shipped in the Kubernetes repositories, it's important to install `kubeadm` manually. The kubeadm
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
Even though `kubeadm` ships in the Kubernetes repositories, it's important to install `kubeadm` manually.
```
So:
> Even though `kubeadm` ships in the Kubernetes repositories, it's important to install `kubeadm` manually.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm
+$ chmod a+rx /usr/bin/kubeadm
+```
+
+**Caution:** Upgrading the `kubeadm` package on your system prior to upgrading the control plane causes a failed upgrade. 
+Even though `kubeadm` is shipped in the Kubernetes repositories, it's important to install `kubeadm` manually. The kubeadm 
+team is working on fixing this limitation. 
+{: .caution}
+
+Verify that this download of kubeadm works, and has the expected version:
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Verify that this download of kubeadm works and has the expected version:

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+```shell
+$ kubeadm version
+```
+
+2. If this the first time you use `kubeadm upgrade`, you will need to create a ConfigMap in order to preserve the 
+configuration for future upgrades. Note that for the below command, you will need to recall what CLI args you
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
2. If this your first time using `kubeadm upgrade`, you need to [create a ConfigMap](/docs/tasks/configure-pod-container/configmap/) in order to preserve the configuration for future upgrades. Note that for the following command, you will need to recall what command line arguments you passed to `kubeadm init` the first time.
```
So:
> 2. If this your first time using `kubeadm upgrade`, you need to [create a ConfigMap](/docs/tasks/configure-pod-container/configmap/) in order to preserve the configuration for future upgrades. Note that for the following command, you will need to recall what command line arguments you passed to `kubeadm init` the first time.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+configuration for future upgrades. Note that for the below command, you will need to recall what CLI args you 
+passed to `kubeadm init` the first time.
+
+If you used flags, do:
+
+```shell
+$ kubeadm config upload from-flags [flags]
+```
+
+Where `flags` can be empty.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
...where `flags` can be empty.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+If you used a config file, do:
+
+```shell
+$ kubeadm config upload from-file --config [config]
+```
+
+Where the `config` is mandatory.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
...where the `config` is mandatory.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+You can now apply the upgrade by executing the following command:
+
+	kubeadm upgrade apply v1.9.0
+
+Note: Before you do can perform this upgrade, you have to update kubeadm to v1.9.0
+
+_____________________________________________________________________
+```
+
+The `kubeadm upgrade plan` checks that your cluster is in an upgradeable state and fetches the versions available to upgrade to in an user-friendly way.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
The `kubeadm upgrade plan` checks that your cluster is upgradeable and fetches the versions available to upgrade to in an user-friendly way.
```
So:
> The `kubeadm upgrade plan` checks that your cluster is upgradeable and fetches the versions available to upgrade to in an user-friendly way.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+
+4. Pick a version to upgrade to and run, for example, `kubeadm upgrade apply` as follows:
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: 4. Pick a version to upgrade to and run. For example:

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+[addons] Applied essential addon: kube-proxy
+
+[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.0". Enjoy!
+
+[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
+```
+
+`kubeadm upgrade apply` does the following:
+
+- It checks that your cluster is in an upgradeable state, that is:
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: +- Checks that your cluster is in an upgradeable state:

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+  - The API Server is reachable,
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
- The API server is reachable
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+  - All nodes are in the `Ready` state, and
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
  - All nodes are in the `Ready` state
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+  - The control plane is healthy
+- It enforces the version skew policies.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
- Enforces the version skew policies
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+- It makes sure the control plane images are available or available to pull to the machine.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
- Makes sure the control plane images are available or available to pull to the machine.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+- It upgrades the control plane components or rollbacks if any of them fails to come up.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
- Upgrades the control plane components or rolls them back if any fail to come up
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+- It applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+
+5. Manually upgrade your Software Defined Network (SDN).
+
+   Your Container Network Interface (CNI) provider might have its own upgrade instructions to follow now.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow.
```
So:
> Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 18]
```diff
@@ -0,0 +1,277 @@
...
+   Check the [addons](/docs/concepts/cluster-administration/addons/) page to
+   find your CNI provider and see if there are additional upgrade steps
+   necessary.
+
+6. Add RBAC permissions for automated certificate rotation. In the future, kubeadm will perform this step automatically:
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
6. Add RBAC permissions for automated certificate rotation.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 19]
```diff
@@ -0,0 +1,277 @@
...
+
+4. After upgrading `kubelet` on each host in your cluster, verify that all nodes are available again by executing the following (from anywhere, for example, from outside the cluster):
+
+```shell
+$ kubectl get nodes
+```
+
+If the `STATUS` column of the above command shows `Ready` for all of your hosts, you are done.
+
+## Recovering from a bad state
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
## Recovering from a failure state
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 20]
```diff
@@ -0,0 +1,277 @@
...
+
+If `kubeadm upgrade` somehow fails and fails to roll back, due to an unexpected shutdown during execution for instance,
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
If `kubeadm upgrade` somehow fails and fails to roll back, for example due to an unexpected shutdown during execution,
```
So:
> If `kubeadm upgrade` somehow fails and fails to roll back, for example due to an unexpected shutdown during execution,

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 21]
```diff
@@ -0,0 +1,277 @@
...
+you may run `kubeadm upgrade` again as it is idempotent and should eventually make sure the actual state is the desired state you are declaring.
```
Comment (zacharysarah) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]:
```
you can run `kubeadm upgrade` again.
```

**Comment by jamiehannaford** [comment]
Thanks for the review @zacharysarah, changes have been pushed.

@luxas ping for tech review

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 22]
```diff
@@ -0,0 +1,275 @@
...
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` Kubernetes cluster running version 1.8.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md) carefully.
+- `kubeadm upgrade` now allows you to upgrade etcd. To do so, use the appropriate flag.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: `kubeadm upgrade` enforces that you upgrade etcd to 3.1.10 as part of upgrading from v1.8 to v1.9. This is due to that etcd 3.1.10 is the officially validated etcd version for Kubernetes v1.9. The upgrade is handled automatically by kubeadm for you.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 23]
```diff
@@ -0,0 +1,275 @@
...
+team is working on fixing this limitation. 
+{: .caution}
+
+Verify that this download of kubeadm works and has the expected version:
+
+```shell
+$ kubeadm version
+```
+
+2. 2. If this your first time using `kubeadm upgrade`, you need to [create a ConfigMap](/docs/tasks/configure-pod-container/configmap/) in order to preserve the configuration for future upgrades. Note that for the following command, you will need to recall what command line arguments you passed to `kubeadm init` the first time.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: is `2. 2.` here right?

Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: This doesn't apply anymore, as all v1.8 and above clusters have the ConfigMap already. This was only a thing when upgrading from v1.7

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md (hunk 18 above)
Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: This is not needed anymore kubeadm does this automatically now

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 24]
```diff
@@ -0,0 +1,275 @@
...
+_____________________________________________________________________
+```
+
+The `kubeadm upgrade plan` checks that your cluster is upgradeable and fetches the versions available to upgrade to in an user-friendly way.
+
+4. Pick a version to upgrade to and run. For example:
+
+```shell
+$ kubeadm upgrade apply v1.9.0
+[preflight] Running pre-flight checks
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: If you try to do a `kubeadm upgrade apply v1.9.0-beta.1` with kubeadm from HEAD locally, I think the output will be a bit different (also considering etcd upgrades etc.). Can you please paste the latest output?

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 25]
```diff
@@ -0,0 +1,275 @@
...
+Kubelet     1 x v1.8.1   v1.8.6
+
+Upgrade to the latest version in the v1.8 series:
+
+COMPONENT            CURRENT   AVAILABLE
+API Server           v1.8.1    v1.8.6
+Controller Manager   v1.8.1    v1.8.6
+Scheduler            v1.8.1    v1.8.6
+Kube Proxy           v1.8.1    v1.8.6
+Kube DNS             1.14.4    1.14.4
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: nit: I think the kube-dns version should be 1.14.5 for 1.8.6

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 26]
```diff
@@ -0,0 +1,275 @@
...
+Kubelet     1 x v1.8.1   v1.9.0
+
+Upgrade to the latest experimental version:
+
+COMPONENT            CURRENT   AVAILABLE
+API Server           v1.8.1    v1.9.0
+Controller Manager   v1.8.1    v1.9.0
+Scheduler            v1.8.1    v1.9.0
+Kube Proxy           v1.8.1    v1.9.0
+Kube DNS             1.14.4    1.14.4
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: nit: I think the kube-dns version for v1.9.0 will be 1.14.7

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 27]
```diff
@@ -0,0 +1,275 @@
+---
+approvers:
+- pipejakob
+- luxas
+- roberthbailey
+- jbeda
+title: Upgrading kubeadm clusters from 1.8 to 1.9
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: We now support downgrades as well; can we rephrase this to be sth like `Upgrading/downgrading kubeadm clusters between v1.8 and v1.9`?

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 28]
```diff
@@ -0,0 +1,275 @@
...
+- `kubeadm upgrade` now allows you to upgrade etcd. To do so, use the appropriate flag.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up what's important to you. For example, any app-level state, such as a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
+
+Also, note that only one minor version upgrade is supported. For example, you can only upgrade from 1.8 to 1.9, not from 1.7 to 1.9.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Upgrading your control plane
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Please add a section for downgrades below as well

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 29]
```diff
@@ -0,0 +1,275 @@
...
+This guide is for upgrading `kubeadm` clusters from version 1.8.x to 1.9.x, as well as 1.8.x to 1.8.y and 1.9.x to 1.9.y where `y > x`.
+See also [upgrading kubeadm clusters from 1.7 to 1.8](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/) if you're on a 1.7 cluster currently.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` Kubernetes cluster running version 1.8.0 or higher in order to use the process described here.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: We could mention again here that swap needs to be disabled, just in case it helps someone.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md (hunk 28 above)
Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: And note that etcd WILL NOT be downgraded if you do v1.8 (3.0.17) -> v1.9 (3.1.10) -> v1.8 (3.1.10) but that it shouldn't technically be a problem

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 30]
```diff
@@ -511,6 +511,8 @@ Instructions for upgrading kubeadm clusters are available for:
  * [1.7.x to 1.7.y  upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
  * [1.7 to 1.8 upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
  * [1.8.x to 1.8.y upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
+ * [1.8 to 1.9 upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/)
```
Comment (luxas) [inline docs/setup/independent/create-cluster-kubeadm.md]: upgrades/downgrades

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md (hunk 24 above)
Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Actually, now did it for you :)

```console
kubeadm upgrade apply v1.9.0
[preflight] Running pre-flight checks.
[upgrade] Making sure the cluster is healthy:
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
[upgrade/version] You have chosen to upgrade to version "v1.9.0"
[upgrade/versions] Cluster version: v1.8.1
[upgrade/versions] kubeadm version: v1.9.0
[upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
[upgrade/prepull] Will prepull images for components [kube-apiserver kube-controller-manager kube-scheduler]
[upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.9.0"...
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests802453804/etcd.yaml"
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/etcd.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests502223003/etcd.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=etcd
[upgrade/staticpods] Component "etcd" upgraded successfully!
[upgrade/staticpods] Writing upgraded Static Pod manifests to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests802453804"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests802453804/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests802453804/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests802453804/kube-scheduler.yaml"
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-apiserver.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests502223003/kube-apiserver.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-apiserver
[upgrade/staticpods] Component "kube-apiserver" upgraded successfully!
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-controller-manager.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests502223003/kube-controller-manager.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-controller-manager
[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests502223003/kube-scheduler.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-scheduler
[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
[bootstraptoken] Configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
[addons] Applied essential addon: kube-dns
[addons] Applied essential addon: kube-proxy

[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.0". Enjoy!

[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md [hunk 31]
```diff
@@ -0,0 +1,275 @@
...
+   Check the [addons](/docs/concepts/cluster-administration/addons/) page to
+   find your CNI provider and see if there are additional upgrade steps
+   necessary.
+
+6. Add RBAC permissions for automated certificate rotation. In the future, kubeadm will perform this step automatically:
+
+```shell
+$ kubectl create clusterrolebinding kubeadm:node-autoapprove-certificate-rotation --clusterrole=system:certificates.k8s.io:certificatesigningrequests:selfnodeclient --group=system:nodes
+```
+
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: We might want to add an "optional features selection during upgrades" or similar section...
And describe that you can set `--feature-gates CoreDNS=true` and that'll swap out kube-dns for CoreDNS for you automatically without downtime and/or enable `--feature-gates SelfHosting=true` which will give you a self-hosted cluster at the same time.

Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: Somewhere here we should also mention certificate rotation and say that if the `/etc/kubernetes/pki/apiserver.crt` cert expires in less than 180 days old when upgrading, kubeadm will move `/etc/kubernetes/pki/apiserver.{crt,key}` to `/etc/kubernetes/pki/expired` and generate a new serving certificate.

**Comment by zacharysarah** [comment]
@jamiehannaford 👋 
- I changed the base to `release-1.9` because docs has a different relationship with `master` and feature branches. 
- Bumping for updates to review feedback from @luxas (thanks!).

**Comment by zacharysarah** [comment]
@jamiehannaford 👋 Bumping for a response to review feedback. Deadline for completion is tomorrow, Friday, December 8th. 

/cc @idvoretskyi

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md (hunk 24 above)
Comment (jamiehannaford) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-9.md]: thanks!

**Comment by jamiehannaford** [comment]
@zacharysarah @luxas All changes incorporated. The only one I didn't have time for is adding a section for downgrades. Is this a blocker or can we add in a PR after the docs freeze? We have most of the content here. I don't mind either way.

**Comment by zacharysarah** [comment]
@jamiehannaford I'm fine with making an update after release. 👍
